public/js/client_registration.js

```diff
@@ -0,0 +1,185 @@
+function addEntry() {
+    var scopes = document.getElementById('scope_input');
+    var container = document.getElementById('scope_container');
+    var errorElement = document.getElementById('Scope');
+
+    if (!scopes.value.trim()) {
+        errorElement.textContent = "Scope cannot be empty";
+        return;
+    }
+    
+    if (scopes.value.includes(' ')) {
+        errorElement.textContent = "Scope cannot contain spaces";
+        return;
+    }
+
+    var wrapper = document.createElement('div');
+    wrapper.classList.add('wrapper');
+
+    var enteredText = document.createElement('div');
+    enteredText.textContent = scopes.value;
+    enteredText.classList.add('entered_text');
+
+    var deleteButton = document.createElement('button');
+    deleteButton.innerHTML = '&times;';
+    deleteButton.classList.add('delete-button');
+    deleteButton.onclick = function () {
+        container.removeChild(wrapper);
+        updateScopeValues();
+        checkScopeError();
+    };
+
+    wrapper.appendChild(enteredText);
+    wrapper.appendChild(deleteButton);
+    container.appendChild(wrapper);
+
+    scopes.value = '';
+    updateScopeValues();
+    checkScopeError();
+}
+
+function updateScopeValues() {
+    var container = document.getElementById('scope_container');
+    var scopeValuesInput = document.getElementById('scope');
+
+    var scopeValues = Array.from(container.querySelectorAll('.entered_text')).map(function (element) {
+        return element.textContent;
+    }).join(', ');
+
+    scopeValuesInput.value = scopeValues;
+}
+
+function checkScopeError() {
+    const scopeErrorElement = document.getElementById("Scope");
+    if (scopeErrorElement && scopeErrorElement.classList.contains("error-message") && scopeErrorElement.textContent.includes("empty")) {
+        scopeErrorElement.textContent = "";
+    }
+}
+
+document.addEventListener("DOMContentLoaded", function () {
+    const form = document.querySelector("form");
+    form.addEventListener("submit", function (event) {
+        event.preventDefault();
+
+        const formData = new FormData(form);
+        fetch("/client-registration", {
+            method: "POST",
+            body: formData
```

Comment:
It does seem that the request includes the credentials by default. Check if using `"credentials": "same-origin"` to only send cookies that originated from our domain is what will improve our security.